lib/deaf_grandma.rb

```diff
@@ -20,8 +20,20 @@ def run!
 
   def speak(input)
 
-    #Implement your code here <<<<<<<<<
-
+	
+	case
+		when input == "BYE"
+			@bye_counter += 1
+		when input == input.upcase
+			return "NOT SINCE 1964!"
+		else
+			return "SPEAK UP SONNY!"
+	end
+
+	if @bye_counter == 3
+		return "SEE YOU LATER SONNY!"
```

Comment:
Per Jay's comment, depending on the editor you're using you should be able to find a setting for how many spaces a tab inserts. The default is usually 4, but you can set it to 2 to meet the conventions of Ruby.